tools/read-dependencies

```diff
@@ -101,6 +101,13 @@ def get_parser():
     parser.add_argument(
         '-d', '--distro', type=str, choices=DISTRO_PKG_TYPE_MAP.keys(),
         help='The name of the distro to generate package deps for.')
+    parser.add_argument(
+        '-b', '--build-requires', action='store_true', default=False,
+        dest='build_requires', help='Print only buildtime required packages')
+    parser.add_argument(
+        '-R', '--runtime-requires', action='store_true', default=False,
+        dest='runtime_requires',
+        help='Print only runtime required packages')
```

Comment:
I find the word *only* a bit confusing here:
 - If both options are specified then the output won't match the option descriptions, as there won't be *only* one dep set being printed
 - If both are specified it's still unclear if *all* the deps are being printed.

Making them mutually exclusive clarifies a bit:

```
    deptype = parser.add_mutually_exclusive_group()
    deptype.add_argument(
        '-R', '--runtime-requires', action='store_true', default=False,
        dest='runtime_requires',
        help='Print only runtime required packages')
    deptype.add_argument(
        '-b', '--build-requires', action='store_true', default=False,
        dest='build_requires', help='Print only buildtime required packages')
```

but better IMHO would be to replace them with a `--distro-deptype` argument with `type=str, choices=['all', 'buildtime', 'runtime'], default='all'` argument. The argument name also suggest that it affects the distro-specific deps.

Reply:
Thanks, I think the mutual exclusion makes sense.  I don't think (especially now that I'm including the python runtime requirements, rather than system runtime packages) the name implies distro specific packages.

Reply:
@raharper my point is that `--runtime-requires`/`--build-requires` have an effect only when you are asking for dependencies as distro packages, specifying `-d <distro>`, as the buildtime/runtime distinction only makes sense when building a distro package. Those options are useless when asking for the Python dependencies.

It would be nice to require `-d` to be specified when using `-b`/`-R`, but this can't be done with pure argparse, it requires a later check.

I am not entirely happy, but I don't think we can really improve on the current situation without a larger refactor, as you wrote above. For the moment I think this PR can land.

Reply:
> @raharper my point is that `--runtime-requires`/`--build-requires` have an effect only when you are asking for dependencies as distro packages, specifying `-d <distro>`, as the buildtime/runtime distinction only makes sense when building a distro package. Those options are useless when asking for the Python dependencies.

s/building a distro package/interacting with distro

It's also useful for --install 

Which, is ambiguous as well as you could pip install or $distro_cmd install 

> 
> It would be nice to require `-d` to be specified when using `-b`/`-R`, but this can't be done with pure argparse, it requires a later check.
> 
> I am not entirely happy, but I don't think we can really improve on the current situation without a larger refactor, as you wrote above. For the moment I think this PR can land.

Thanks.  It could use a refactor, but this path is primarily used for building the distro packages; so fixing it's bugs is step 1.